Linked Accounts/Link-Accounts.ps1

```diff
@@ -108,37 +98,49 @@ Function ConvertTo-URL($sText)
 .PARAMETER sText
 	The text to encode
 #>
-	if (![string]::IsNullOrEmpty($sText))
-	{
+	if (![string]::IsNullOrEmpty($sText)) {
 		Write-Debug "Returning URL Encode of $sText"
 		return [URI]::EscapeDataString($sText)
-	}
-	else
-	{
+	} else {
 		return $sText
 	}
 }
+Function Compare-Version () {
```

Comment:
@bab29 , Would you consider using a method I built for this exact use case?

# @FUNCTION@ ======================================================================================================================
# Name...........: Test-RESTVersion
# Description....: Tests if the requested version exists in the PVWA REST API
# Parameters.....: version
# Return Values..: True / False
# =================================================================================================================================
Function Test-RESTVersion
{
<#
.SYNOPSIS
	Tests if the requested version exists in the PVWA REST API
.DESCRIPTION
	Tests if the requested version exists in the PVWA REST API
.PARAMETER Version
	A string of the requested PVWA REST version to test
#>

    param (
		[Parameter(Mandatory=$true)]
		[string]$version
	)

	$retVersionExists = $false
	try{
		Write-LogMessage -Type debug -Msg "Testing to see if the PVWA is at least in version $version"
		$serverResponse = Invoke-REST -Command GET -URI $URL_Server
		if($null -ne $serverResponse)
		{
			Write-LogMessage -Type debug -Msg "The current PVWA is in version $($serverResponse.ExternalVersion)"
			$majorVersionCompare = ( [int]($serverResponse.ExternalVersion.Split('.')[0]) -gt [int]($version.Split('.')[0]) )
			$majorVersionEqual = ( [int]($serverResponse.ExternalVersion.Split('.')[0]) -eq [int]($version.Split('.')[0]) )
			$minorVersionCompare = ( [int]($serverResponse.ExternalVersion.Split('.')[1]) -ge [int]($version.Split('.')[1]) )
			if($majorVersionCompare -or ($majorVersionEqual -and $minorVersionCompare))
			{
				$retVersionExists = $true
			}
		}
		else
		{
			Throw "An error occurred while testing the PVWA version"
		}

		return $retVersionExists
	} catch {
		# Check the error code returned from the REST call
		$innerExcp = $_.Exception.InnerException
		Write-LogMessage -Type Verbose -Msg "Status Code: $($innerExcp.StatusCode); Status Description: $($innerExcp.StatusDescription); REST Error: $($innerExcp.CyberArkErrorMessage)"
		if($innerExcp.StatusCode -eq "NotFound") {
			return $false
		}
		else{
			Throw $(New-Object System.Exception ("Test-RESTVersion: There was an error checking for REST version $version.",$_.Exception))
		}
	}
}

Reply:
You use it the same way:
If (Test-RESTVersion -version "10.4")
{...}

Reply:
Sure, I will make this update.

Reply:
I added this but isn't this easier then doing all the splits adn compare?

If ([version]($serverResponse.InternalVersion) -ge [version]$version) {$retVersionExists = $true}